packages/ai/core/generate-text/stream-text.test.ts

```diff
@@ -3415,59 +3415,60 @@ describe('streamText', () => {
 
   describe('options.transform', () => {
     describe('with base transformation', () => {
-      const upperCaseTransform =
-        <TOOLS extends ToolSet>() =>
-        (options: { tools: TOOLS }) =>
-          new TransformStream<TextStreamPart<TOOLS>, TextStreamPart<TOOLS>>({
-            transform(chunk, controller) {
-              if (chunk.type === 'text-delta') {
-                chunk.textDelta = chunk.textDelta.toUpperCase();
-              }
+      const upperCaseTransform = () =>
+        new TransformStream<
+          TextStreamPart<{ tool1: Tool<{ value: string }> }>,
+          TextStreamPart<{ tool1: Tool<{ value: string }> }>
```

Comment:
why does the inference no longer work? think i did the double functions on purpose back then

Reply:
now the types are better because they aren't just any